Find a functionally equivalent code:android.net.wifi.p2p.WifiP2pManager.initialize:COMMENT
Method Modifier: public      
Comment:/**
 * Registers the application with the Wi-Fi framework. This function
 * must be the first to be called before any p2p operations are performed.
 *
 * @param srcContext is the context of the source
 * @param srcLooper is the Looper on which the callbacks are receivied
 * @param listener for callback at loss of framework communication. Can be null.
 * @return Channel instance that is necessary for performing any further p2p operations
 */

@@ -1,11 +1,3 @@
 {
-    Messenger messenger = getMessenger();
-    if (messenger == null)
-        return null;
-    Channel c = new Channel(srcContext, srcLooper, listener);
-    if (c.mAsyncChannel.connectSync(srcContext, c.mHandler, messenger) == AsyncChannel.STATUS_SUCCESSFUL) {
-        return c;
-    } else {
-        return null;
-    }
+    return initalizeChannel(srcContext, srcLooper, listener, getMessenger());
 }

